clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -383,6 +390,33 @@ class CIRGenBuilderTy : public mlir::OpBuilder {
     return Address(baseAddr, ptrTy, addr.getAlignment());
   }
 
+  // FIXME(cir): this class should have an attribute with a reference to the
+  // module so that we don't have search for it or pass it around. Also, it
+  // would be good to track a list of globals, or at least the last one
+  // inserted, so that we can insert globals in the same order they are defined
+  // by CIRGen.
+  mlir::cir::GlobalOp
+  createVersionedGlobal(mlir::ModuleOp module, mlir::Location loc,
+                        mlir::StringRef name, mlir::Type type, bool isConst,
+                        mlir::cir::GlobalLinkageKind linkage) {
+    mlir::OpBuilder::InsertionGuard guard(*this);
+    setInsertionPointToStart(module.getBody());
+
+    // Create a unique name if it already exists.
+    std::string uniqueName;
+    if (unsigned version = GlobalsVersioning[name.str()]++)
+      uniqueName = name.str() + "." + std::to_string(version);
+    else
+      uniqueName = name.str();
+
+    return create<mlir::cir::GlobalOp>(loc, uniqueName, type, isConst, linkage);
+  }
+
+  mlir::Value createGetGlobal(mlir::cir::GlobalOp global) {
```

Comment:
If you are adding this helper, can you please replace all other adhoc's `create<mlir::cir::GetGlobalOp>` to use this? Follow up PR sounds good.